openrtb_ext/utils.go

```diff
@@ -0,0 +1,91 @@
+package openrtb_ext
+
+import (
+    "github.com/mxmCherry/openrtb"
+    "encoding/json"
+    "math/rand"
+    "fmt"
+)
+
+// Quick little randomizer for a list of strings. Stuffing it in utils to keep other files clean
+func RandomizeList(list []BidderName) {
+    l := len(list)
+    perm := rand.Perm(l)
+    var j int
+    for i :=0; i < l; i++ {
+        j = perm[i]
+        list[i], list[j] = list[j], list[i]
+    }
+}
+
+// This will copy the openrtb BidRequest into an array of requests, where the BidRequest.Imp[].Ext field will only
+// consist of the "prebid" field and the field for the appropriate bidder parameters. We will drop all extended fields
+// beyond this context, so this will not be compatible with any other uses of the extension area. That is, this routine
+// will work, but the adapters will not see any other extension fields.
+
+// NOTE: the return map will only contain entries for bidders that both have the extension field in at least one Imp,
+// and are listed in the adapters string. wseats and bseats can be implimented by passing the bseats list as adapters,
+// or after return removing any adapters listed in wseats. Or removing all adapters in wseats from the adapters list
+// before submitting.
+
+// Take an openrtb request, and a list of bidders, and return an openrtb request sanitized for each bidder
+func CleanOpenRTBRequests(orig *openrtb.BidRequest, adapters []BidderName) (map[BidderName]*openrtb.BidRequest, []error) {
+    // This is the clean array of openrtb requests we will be returning
+    cleanReqs := map[BidderName]*openrtb.BidRequest{}
+    errList := make([]error, 0, 1)
+
+    // Decode the Imp extensions once to save time. We store the results here
```

Comment:
This would make a good separate function, since it does a standalone job.

Reply:
Which part would be a stand alone function? Decoding the incoming Imp extensions?

Reply:
yeah. Lines 39 - 47 look like a standalone`func(imps []openrtb.Imp) []map[string]interface{}`